server/Mistakes.Journal.Api/Api/Mistakes/Controllers/MistakesController.cs

```diff
@@ -24,20 +27,39 @@ MistakesJournalContext dataContext
             _dataContext = dataContext;
         }
 
+        #endregion
+
         #region POST
 
         [HttpPost]
         public async Task<ActionResult<MistakeWebModel>> AddMistake(NewMistakeWebModel newMistake)
         {
             if (!ModelState.IsValid)
                 return BadRequest(ModelState);
-            
-            var mistake = new Mistake(newMistake.Name, newMistake.Goal, newMistake.Priority.GetValueOrDefault());
-            mistake.Tips.AddRange(newMistake.Tips.Where(t => !string.IsNullOrWhiteSpace(t)).Select(t => new Tip(t)));
+
+            var mistake = newMistake.ToEntity();
+
+            mistake.Repetitions.Add(new Repetition(newMistake.AddDateTime ?? DateTime.Now));
+
+            if (newMistake.Tips != null)
+                mistake.Tips.AddRange(newMistake.Tips.Where(t => t.IsPresent()).Select(t => new Tip(t)));
+
+            foreach (var labelName in newMistake.Labels.EmptyIfNull())
```

Comment:
I think that frontend might change its representation and send label IDs, instead of names :)